browser/ethereum_remote_client/ethereum_remote_client_service.cc

```diff
@@ -349,15 +346,14 @@ void EthereumRemoteClientService::OnExtensionUninstalled(
     extensions::UninstallReason reason) {
   if (extension->id() == metamask_extension_id) {
     PrefService* prefs = user_prefs::UserPrefs::Get(context_);
-    auto provider = static_cast<brave_wallet::Web3ProviderTypes>(
-        prefs->GetInteger(kBraveWalletWeb3Provider));
-    if (provider == brave_wallet::Web3ProviderTypes::METAMASK)
+    auto provider = brave_wallet::GetDefaultWallet(prefs);
+    if (provider == brave_wallet::mojom::DefaultWallet::Metamask)
       prefs->SetInteger(
```

Comment:
seems like `brave_wallet::SetDefaultWallet` is more appropriate here.

Reply:
Fixed here: 7f0b2f0ffd0c79655515438bd5b6dddeccd59968